eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/EventGridSubscriber.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.eventgrid.customization;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.eventgrid.models.EventGridEvent;
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.serializer.AzureJacksonAdapter;
+import com.microsoft.rest.protocol.SerializerAdapter;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * The type that can be used to de-serialize events.
+ */
+@Beta
+public class EventGridSubscriber {
+    /**
+     * The default adapter for to be used for de-serializing the events
+     */
+    private final AzureJacksonAdapter defaultSerializerAdapter;
+    /**
+     * The map containing user defined mapping of eventType to Java model type
+     */
+    private Map<String, Type> eventTypeToEventDataMapping;
+
+    @Beta
+    public EventGridSubscriber() {
+        this.defaultSerializerAdapter = new AzureJacksonAdapter();
+        this.eventTypeToEventDataMapping =  new HashMap<>();
+    }
+
+    /**
+     * Add a custom event mapping. If a mapping with same eventType exists then the old eventDataType is replaced by
+     * the specified eventDataType
```

Comment:
Nit: Please add a fullstop to all doc comments (assuming documentation will be generated from these comments).

Reply:
done